fspiop-api/documents/Tracing v1.0.md

```diff
@@ -0,0 +1,221 @@
+
+3.2.2. Generate the _tracestate_ header, value below is for example purposes only as it is vendor specific: 
+   > 3.2.2.1. Add a _list-member-keyvalue_ ([Table 4](#table-4)) with value `{vendor-name}=BASE64Encoded({parent-id})`, where _vendor-name_ ([Table 4](#table-4)) is the FSP or Hub identifier and _vendor-value_ ([Table 4](#table-4)) being `BASE64Encoded({parent-id})` with the operation to base 64 encode the _parent-id_ ([Table 2](#table-2)). Base64 encoding the list member value is optional but required if its contents is not printable ASCII<sup>[1](https://www.w3.org/TR/trace-context-1/#value)</sup>.
+
+### 3.3 Continuing a Trace
+
+To create the a trace, the following steps are performed:
+
+3.3.1. Mutate the _traceparent_ ([Table 1](#table-1)) header with the following values:
+   > 3.3.1.1. _version_ ([Table 2](#table-2)) is unchanged.<br/>
+   > 3.3.1.2. _trace-id_ ([Table 2](#table-2)) is unchanged.<br/>
+   > 3.3.1.3. Generate a _parent-id_ ([Table 2](#table-2)) to represent a distribute portion of the trace.<br/>
+   > 3.3.2.4 Set _trace-flags_ ([Table 2](#table-2)) to `01` to indicate if the trace was sampled by the source, otherwise to `00`.<br/>
+
+3.3.2. Mutate the _tracestate_ ([Table 1](#table-1)) header with the following:
+   > 3.3.2.1. Add or update (if it already exists) list member `{VENDOR}=BASE64Encoded({parent-id})`, where `BASE64Encoded({string-value})` is the operation to base 64 encode the input `string-value` to the _tracestate_ header field. Ensure that the added or updated list member is inserted/moved to the front<sup>[1](https://www.w3.org/TR/trace-context-1/#value)</sup> of the list.
+   > 3.3.2.1. Ensure _tracestate_ is less than 512 characters. If necessary remove the last list members until the 512 character limit has been reached <sup>[1](https://www.w3.org/TR/trace-context-1/#tracestate-limits)</sup>.
+
+### 3.4 Forwarding a Trace
+
+Vendors receiving trace request headers ([Table 1](#table-1)) do not need to participate but they MUST forward it in outgoing requests unmodified.
+
+Unmodified header propagation is typically implemented in pass-through services like proxies. As such, this behavior <sup>[1](https://www.w3.org/TR/trace-context-1/#mutating-the-traceparent-field)</sup> may also be implemented in a service which currently does not collect distributed tracing information.
+
+### 4. Examples
+
+#### 4.1. POST /transfers request being sent by FSP1 to FSP2 through a Mojaloop Switch
+
+This example will result in a single end-to-end trace graph being created from `FSP1` to `FSP2` being routed through a `Mojaloop Switch` for a  **POST /transfers** linking the resulting **PUT /transfers** callback from `FSP2` to `FSP1` via the `Mojaloop Switch`.
+
+<br/>
+4.1.1. `FSP1` sends a **POST /transfers** to a `Mojaloop Switch` which includes the following generated trace information:<br/><br/>
```

Comment:
Can we please add some information in the example on how the trace information is generated in the `FSP1`? For example, does the tracestate t61rcWkgMzE mean something for `FSP1`?

Reply:
I have updated the examples to include more information on how IDs were generated.